File: Java/lesson 02/OddOccurrencesInArray.java
// you can also use imports, for example:
// import java.util.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

class Solution {
    public int solution(int[] A) {
        // write your code in Java SE 8
        int result = 0;
        for (int val: A) {
            result ^= val;
        }
        return result;
    }
}

File: JavaScript/lesson 02/OddOccurrencesInArray.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    var values = new Set();
    for (var i = 0; i < A.length; i++) {
        var value = A[i];
        if (values.has(value)) {
            values.delete(value);
        } else {
            values.add(value);
        }
    }
    var iterator = values.keys();
    return iterator.next().value;
}